docroot/modules/custom/civictheme_govcms/src/Govcms/GovcmsManager.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+namespace Drupal\civictheme_govcms\Govcms;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a civictheme_govcms govcms manager.
+ */
+class GovcmsManager {
+
+  /**
+   * The list of removal configurations(methods).
+   *
+   * @var array
+   */
+  protected $removalList;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a GovcmsManager Manager object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager
+  ) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->removalList = [
+      'removeGovcmsMediaTypes',
+      'removeGovcmsTextFormat',
+      'removeGovcmsFields',
+      'removeGovcmsContentTypes',
+      'removeGovcmsVocabularies',
+      'removeGovcmsUserRoles',
+      'removeGovcmsMenus',
+      'removeGovcmsPathautoPatterns',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+    $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Removes the requested config.
+   *
+   * @param string $preserve
+   *   The list of config types to preserve.
+   */
+  public function civicthemeGovcmsRemoveConfig(string $preserve = '') {
+    $preserve_list = [];
+    if (!empty($preserve)) {
+      $preserve_list = array_map(function ($val) {
```

Comment:
what you did here is a good approach.

however we needs a more granular control.

can we please change this to use a map of callbacks

```
[
  // item name in CLI => callable
   'media_types' => [static:class, 'removeMediaTypes'],
]
```

Reply:
Updated the list to Callbacks
https://github.com/salsadigitalauorg/civictheme_source/pull/482/files#diff-a5e4bbe5e1d6eb2e3422122ce3a880c845d4de974e17985192afa3734e1f3564R41